sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/ManagedServiceIdentity.Serialization.cs

```diff
@@ -70,7 +71,7 @@ internal static ManagedServiceIdentity DeserializeManagedServiceIdentity(JsonEle
                 }
                 if (property.NameEquals("type"u8))
                 {
-                    type = JsonSerializer.Deserialize<ManagedServiceIdentityType>("{"+property.ToString()+"}", options);
+                    type = JsonSerializer.Deserialize<ManagedServiceIdentityType>(property.Value.GetRawText(), options);
```

Comment:
we should not use `property.Value.GetString()`.
When the value of this property is foo, `property.Value.GetString()` will return `foo` without quotes which is NOT valid JSON, and `property.Value.GetRawText()` will return `"foo"` with quote which is valid JSON